org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/GradlePreferenceChangeListener.java

```diff
@@ -0,0 +1,136 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Red Hat Inc. and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Red Hat Inc. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jdt.ls.core.internal.managers;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import org.eclipse.buildship.core.GradleDistribution;
+import org.eclipse.buildship.core.WrapperGradleDistribution;
+import org.eclipse.buildship.core.internal.CorePlugin;
+import org.eclipse.buildship.core.internal.configuration.ProjectConfiguration;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.jdt.ls.core.internal.JavaLanguageServerPlugin;
+import org.eclipse.jdt.ls.core.internal.ProjectUtils;
+import org.eclipse.jdt.ls.core.internal.preferences.IPreferencesChangeListener;
+import org.eclipse.jdt.ls.core.internal.preferences.PreferenceManager;
+import org.eclipse.jdt.ls.core.internal.preferences.Preferences;
+import org.eclipse.jdt.ls.internal.gradle.checksums.ValidationResult;
+import org.eclipse.jdt.ls.internal.gradle.checksums.WrapperValidator;
+
+/**
+ * @author snjeza
+ *
+ */
+public class GradlePreferenceChangeListener implements IPreferencesChangeListener {
+	@Override
+	public void preferencesChange(Preferences oldPreferences, Preferences newPreferences) {
+		ProjectsManager projectsManager = JavaLanguageServerPlugin.getProjectsManager();
+		if (projectsManager != null) {
+			if (hasAllowedChecksumsChanged(oldPreferences, newPreferences)) {
+				for (IProject project : ProjectUtils.getGradleProjects()) {
+					updateProject(projectsManager, project);
+				}
+			}
+		}
+	}
+
+	private void updateProject(ProjectsManager projectsManager, IProject project) {
+		String projectDir = project.getLocation().toFile().getAbsolutePath();
+		Path projectPath = Paths.get(projectDir);
+		PreferenceManager preferencesManager = getPreferenceManager();
```

Comment:
no need, you already have access to newPreferences

Reply:
Fixed.